File: 1 Iterations/BinaryGap/binaryGap.js
function solution(N) {
    const a = N.toString(2).split("");
    const counts = new Set();
    let zero = 0;
    for (let y = 0; y < a.length; y += 1) {
      if (a[y] == 0) zero++;
      if (a[y] == 1) {
        counts.add(zero);
        zero = 0;
      }
    }
    return (counts.size == 0) ? 0 : Math.max(...counts);
  }

  console.log(solution(1041)); // 5
  console.log(solution(32)); // 0

File: 1 Iterations/BinaryGap/binaryGap.cpp
#include <set>
#include <string>
#include <vector>
int solution(int N) {
  int bin;
  vector<int> b;
  set<int> counts;
  int zero = 0;
  bool gap = false;

  while (N > 0) {
    bin = N % 2;
    b.push_back(bin);
    N /= 2;
  }

  // for (int x : b) { cout << x; } print binary representation of number vector

  for (unsigned int y = 0; y < b.size(); y += 1) {
    if (b.at(y) == 0 && gap == true) {
      zero++;
    }
    if (b.at(y) == 1) {
      gap = true;
      counts.insert(zero);
      zero = 0;
    }
  }
  // for (int i : counts) { cout << i << ' '; }  print gap counts set
  return (counts.size() == 0) ? 0 : *counts.rbegin(); 
}
